commands/command_push.go

```diff
@@ -61,22 +66,11 @@ func uploadsBetweenRefAndRemote(remote string, refs []string) *lfs.TransferQueue
 			Panic(err, "Error scanning for Git LFS files in the %q ref", ref)
 		}
 
-		for _, p := range pointers {
-			oidPointerMap[p.Oid] = p
-		}
-	}
-
-	i := 0
-	pointers := make([]*lfs.WrappedPointer, len(oidPointerMap))
-	for _, pointer := range oidPointerMap {
-		pointers[i] = pointer
-		i += 1
+		uploadPointers(pointers)
```

Comment:
This going to create a bit more traffic on the API when pushing multiple refs which share unpushed ancestry, and especially so in the --all case. If you take the case where there's a long history and 4 branches in the last few commits, all the oids in the history they shared would be submitted again for each branch. The API would only return links for those that hadn't been completed in the last branch, but still, it's increasing the load quite a bit.